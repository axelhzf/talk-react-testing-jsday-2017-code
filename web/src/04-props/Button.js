import * as React from 'react';
import PropTypes from 'prop-types';

export class Button extends React.Component {
  static propTypes = {
    disabled: PropTypes.bool,
    label: PropTypes.string
  }

  render () {
    const { disabled, label } = this.props;
    return <button disabled={disabled}>{label}</button>
  }
}
