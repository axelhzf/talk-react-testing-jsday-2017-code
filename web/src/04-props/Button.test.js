import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Button } from './Button';

describe('04-props', () => {
  it('should render a button', () => {
    const renderer = ReactTestRenderer.create(<Button label="Click me"/>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render a disabled button', () => {
    const renderer = ReactTestRenderer.create(<Button label="Disabled" disabled/>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});