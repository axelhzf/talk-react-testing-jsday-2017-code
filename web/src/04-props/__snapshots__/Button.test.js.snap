// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`04-props should render a button 1`] = `
<button
  disabled={undefined}
>
  Click me
</button>
`;

exports[`04-props should render a disabled button 1`] = `
<button
  disabled={true}
>
  Disabled
</button>
`;
